mbtools/validate_mbz_html.py

```diff
@@ -119,6 +124,40 @@ def run_validations(html_elements, include_styles):
     return violations
 
 
+def find_qbank_uuid_violations(question_bank):
+    questions = question_bank.latest_questions
+    qbe_to_uuid = {}
+    frequency_dict = {}
+    violations_list = []
+    for question in questions:
+        qbe_to_uuid[question.question_bank_entry_id] = question.id_number
+
+    for qbe_id, id_number in qbe_to_uuid.items():
+        if not utils.validate_uuid4(id_number):
+            violations_list.append(Violation(INVALID_QBANK_UUID_VIOLATION,
+                                             question_bank.questionbank_path,
+                                             f'question id: {qbe_id} uuid: '
+                                             f'{id_number}'))
+        if id_number in frequency_dict:
+            frequency_dict[id_number] += 1
+        else:
+            frequency_dict[id_number] = 1
+        if frequency_dict[id_number] > 1:
```

Comment:
This approach works, but we seem to be maintaining a count that never gets used. An alternative could be to compile a set in something like `observed_ids` or something like that, and then just do something like:

```python
    for qbe_id, id_number in qbe_to_uuid.items():
        if not utils.validate_uuid4(id_number):
            violations_list.append(Violation(INVALID_QBANK_UUID_VIOLATION,
                                             question_bank.questionbank_path,
                                             f'question id: {qbe_id} uuid: '
                                             f'{id_number}'))
        if id_number in observed_ids:
            violations_list.append(Violation(DUPLICATE_QBANK_UUID_VIOLATION,
                                             question_bank.questionbank_path,
                                             f'question id: {qbe_id} uuid: '
                                             f'{id_number}'))
        observed_ids.add(id_number)
```

It's subjective, but :point_up: might be a bit simpler :man_shrugging: .